src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,109 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
+ * - Tracking when the user manipulates a future value
+ * - Propagating current values to future values if they
+ * haven't been manipulated by the user
+ * - Making this instance serializable
+ * - Making sure only valid values are saved.
+ * 
+ * All values stored in a FormValue instance should be valid
+ * 
+ * There must be a better way to write that.
+ * 
+ * @class
+ * @param {string} name - the name of the client property
+ * @param {any} [val] - the initial value
+ * @param {function} [makeValid] - returns a valid value no
+ * matter what. Will be given the new value, the current value,
+ * the string 'current' or 'future', and the `FormValue` instance.
+ * 
+ * @returns FormValue
+ * @todo Finish description of FormValue return value
+ */
+class FormValue {
+
+  constructor ( name, val = null, makeValid ) {
+
+    var fVal = this;
```

Comment:
Why not just use `this` in the following variable assignments?

Reply:
Managing `this` is a whole discussion we can have sometime. Let's see if we can clear some distractions this Tuesday and get into it.

Reply:
Sure. Most of the time, people save to var when passing to a callback though - it's just a bit odd when we're not doing anything with it.